validator/client/validator.go

```diff
@@ -1139,7 +1140,38 @@ func (v *validator) StartEventStream(ctx context.Context, topics []string, event
 	v.validatorClient.StartEventStream(ctx, topics, eventsChannel)
 }
 
-func (v *validator) ProcessEvent(event *eventClient.Event) {
+func (v *validator) checkDependentRoots(ctx context.Context, head *structs.HeadEvent, slot primitives.Slot) error {
+	if head == nil {
+		return errors.New("received empty head event")
+	}
+	prevDepedentRoot, err := bytesutil.DecodeHexWithLength(head.PreviousDutyDependentRoot, fieldparams.RootLength)
+	if err != nil {
+		return errors.Wrap(err, "failed to decode previous duty dependent root")
+	}
+	currEpochStart, err := slots.EpochStart(slots.ToEpoch(slot))
+	if err != nil {
+		return err
+	}
+	if !bytes.Equal(prevDepedentRoot, v.duties.PrevDependentRoot) {
+		if err := v.UpdateDuties(ctx, currEpochStart); err != nil {
```

Comment:
Also this is wrong, you **will not** update your duties. This is the first line in `UpdateDuties` 

```go
	if !slots.IsEpochStart(slot) && v.duties != nil {
		// Do nothing if not epoch start AND assignments already exist.
		return nil
	}
```

Reply:
It's passing in `currEpochStart` which satisfy `IsEpochStart` right

Reply:
I misread, ignore the comment